superset-frontend/src/explore/components/controls/MetricsControl.jsx

```diff
@@ -103,58 +109,25 @@ function coerceAdhocMetrics(value) {
   });
 }
 
-function getDefaultAggregateForColumn(column) {
-  const { type } = column;
-  if (typeof type !== 'string') {
-    return AGGREGATES.COUNT;
-  }
-  if (type === '' || type === 'expression') {
-    return AGGREGATES.SUM;
-  }
-  if (
-    type.match(/.*char.*/i) ||
-    type.match(/string.*/i) ||
-    type.match(/.*text.*/i)
-  ) {
-    return AGGREGATES.COUNT_DISTINCT;
-  }
-  if (
-    type.match(/.*int.*/i) ||
-    type === 'LONG' ||
-    type === 'DOUBLE' ||
-    type === 'FLOAT'
-  ) {
-    return AGGREGATES.SUM;
-  }
-  if (type.match(/.*bool.*/i)) {
-    return AGGREGATES.MAX;
-  }
-  if (type.match(/.*time.*/i)) {
-    return AGGREGATES.COUNT;
-  }
-  if (type.match(/unknown/i)) {
-    return AGGREGATES.COUNT;
-  }
-  return null;
-}
```

Comment:
I wonder if we could still leverage this logic for setting the default aggregate if it's unset and the user chooses the column first (the usual case)?